src/components/header/EmptyThreadIndicator.js

```diff
@@ -126,4 +75,72 @@ class EmptyThreadIndicator extends PureComponent<Props> {
   }
 }
 
+/**
+ * Define this outside of the class so that it's easily testable. The internals
```

Comment:
The math was defying labeling, so rather than over-complicate things with crazy names, I put in some solid unit tests to make sure the behavior is correct.